### PR TITLE
[Fundamental] Limit output tracebacks and add documentation on rotate aoai keys

### PR DESCRIPTION
**Pull request by crazygao**
# Description

Limit output tracebacks

Reference https://docs.pytest.org/en/latest/how-to/output.html#modifying-python-traceback-printing

# All Promptflow Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes].**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](../CONTRIBUTING.md).**
- [ ] **Create an issue and link to the pull request to get dedicated review from promptflow team. Learn more: [suggested workflow](../CONTRIBUTING.md#suggested-workflow).**

## General Guidelines and Best Practices
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### Testing Guidelines
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: .github/workflows/promptflow-core-test.yml [hunk 1]
```diff
@@ -71,7 +71,7 @@ jobs:
         run: |
           cp ${{ github.workspace }}/src/promptflow/dev-connections.json.example ${{ github.workspace }}/src/promptflow/connections.json
       - name: run core tests
-        run: poetry run pytest ./tests/core --cov=promptflow --cov-config=pyproject.toml --cov-report=term --cov-report=html --cov-report=xml
+        run: poetry run pytest ./tests/core --cov=promptflow --cov-config=pyproject.toml --cov-report=term --cov-report=html --cov-report=xml --tb=short
```
Comment (zhengfeiwang) [inline .github/workflows/promptflow-core-test.yml:74]: may add how `--tb=short` behaves in the PR description

Comment (zhengfeiwang) [reply .github/workflows/promptflow-core-test.yml:74]: also, will `short` resolve the leak issue? shall we use `no`?

Comment (crazygao) [reply .github/workflows/promptflow-core-test.yml:74]: Let's try short first https://docs.pytest.org/en/latest/how-to/output.html#modifying-python-traceback-printing

Comment (zhengfeiwang) [reply .github/workflows/promptflow-core-test.yml:74]: verified, `short` works well

image

**Review by zhengfeiwang — APPROVED** [review]

**Review by brynn-code — APPROVED** [review]

**Pull request merged**